File: extensions/shared-youtube/library/src/main/java/app/silva/extension/shared/patches/DisableQUICProtocolPatch.java
package app.silva.extension.shared.patches;

import app.silva.extension.shared.settings.SharedYouTubeSettings;

@SuppressWarnings("unused")
public class DisableQUICProtocolPatch {

    public static boolean disableQUICProtocol(boolean original) {
        boolean isDisabled = SharedYouTubeSettings.DISABLE_QUIC_PROTOCOL.get();
        return !isDisabled && original;
    }
}

File: patches/src/main/kotlin/app/silva/patches/youtube/misc/quic/DisableQUICProtocolPatch.kt
package app.silva.patches.youtube.misc.quic

import app.silva.patches.shared.misc.quic.disableQUICProtocolPatch
import app.silva.patches.youtube.misc.extension.sharedExtensionPatch
import app.silva.patches.youtube.misc.settings.PreferenceScreen
import app.silva.patches.youtube.misc.settings.settingsPatch
import app.silva.patches.youtube.shared.Constants.COMPATIBILITY_YOUTUBE

@Suppress("unused")
val disableQUICProtocolPatchYouTube = disableQUICProtocolPatch(
    block = {
        dependsOn(
            sharedExtensionPatch,
            settingsPatch
        )

        compatibleWith(COMPATIBILITY_YOUTUBE)
    },
    preferenceScreen = PreferenceScreen.MISC
)